model/data/recipient.h

```diff
@@ -12,35 +14,47 @@ struct Recipient {
         to,
         cc,
         bcc
-    } type;
-    Email email;
-};
+    } type_;
+    Email email_;
 
-inline bool operator < (const Recipient& l, const Recipient& r) {
-    return l.type != r.type ? l.type < r.type : l.email.address < r.email.address;
-}
+    std::string type() const {
+        using RT = Recipient::Type;
+        switch (type_) {
+            case RT::from : return "from";
+            case RT::to : return "to";
+            case RT::cc : return "cc";
+            case RT::bcc : return "bcc";
+            case RT::unknown: break;
+        }
+        return "unknown";
+    }
 
-// These two functions are really pain - it would be nice to have a mechanism
-// to do not write such functions
-inline std::string toString(Recipient::Type t) {
-    using RT = Recipient::Type;
-    switch (t) {
-        case RT::from : return "from";
-        case RT::to : return "to";
-        case RT::cc : return "cc";
-        case RT::bcc : return "bcc";
-        case RT::unknown: break;
+    void setType(const std::string& s) {
+        using RT = Recipient::Type;
+        std::map<std::string, Recipient::Type> str2type = {
+            {"from", RT::from},
+            {"to", RT::to},
+            {"cc", RT::cc},
+            {"bcc", RT::bcc}
+        };
+        if( str2type.count(s) ) {
+            type_ = str2type.at(s);
+        }
+        type_ = RT::unknown;
     }
```

Comment:
I don't think that it is good idea to use the std::map like this. You find element twice in positive case. Construct map every function call. And I think it is slower than the original solution. :)